flagutil/flagutil.go

```diff
@@ -0,0 +1,73 @@
+// Package containing flag utilities. This includes implementations of flag.Value and flag.Getter interfaces. Also
+// contains methods like ValidateFlags.
+package flagutil
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"strings"
+)
+
+// Validates flags and logs a fatal message with all errors if found. Must be called after flag.Parse().
+// This is done by calling Get() on all flag values which implement the flag.Getter interface and collecting errors if returned.
+func ValidateFlags() {
+	var flagErrors []error
+	checkGetError := func(f *flag.Flag) {
+		if fv, ok := f.Value.(flag.Getter); ok {
+			if err, ok := fv.Get().(error); ok && err != nil {
+				flagErrors = append(flagErrors, flagError(fmt.Sprintf("Flag: %s, Error: %s, Usage: %s", f.Name, err.Error(), f.Usage)))
+			}
+		}
+	}
+	flag.VisitAll(checkGetError)
+	if len(flagErrors) > 0 {
+		errorstr := []string{"Errors found while validating flags"}
+		for _, err := range flagErrors {
+			errorstr = append(errorstr, err.Error())
+		}
+		log.Fatal(strings.Join(errorstr, "\n"))
+	}
+}
+
+type flagError string
+
+func (err flagError) Error() string {
+	return string(err)
+}
```

Comment:
The previous approach was to log.Fatal in which case it did not make sense for the error to be exported. But this, in conjunction with returning the error from ValidateFlags, makes more sense. Done in adf31e7.